hocuspocus-server/src/rejectUnauthorized.ts

```diff
@@ -0,0 +1,130 @@
+import type { CloseEvent } from "@hocuspocus/common";
+import {
+  beforeHandleMessagePayload,
+  Extension,
+  IncomingMessage,
+  MessageType,
+} from "@hocuspocus/server";
+import * as syncProtocol from "y-protocols/sync";
+import * as Y from "yjs";
+
+/**
+ * This extension rejects any changes to the restricted type.
+ *
+ * It does this by:
+ * - extracting the yjsUpdate from the incoming message
+ * - applying the update to the restricted type
+ * - if the update is rejected, we throw an error and close the connection
+ * - if the update is accepted, we do nothing
+ */
+export class RejectUnauthorized implements Extension {
+  constructor(private readonly threadsMapKey: string) {}
+  /**
+   * Extract the yjsUpdate from the incoming message
+   * @param message
+   * @returns
+   */
+  private getYUpdate(message: Uint8Array) {
+    /**
+     * The messages we are interested in are of the following format:
+     * [docIdLength: number, ...docIdString: string, hocuspocusMessageType: number,  ySyncMessageType: number, ...yjsUpdate: Uint8Array]
+     *
+     * We check that the hocuspocusMessageType is Sync and that the ySyncMessageType is messageYjsUpdate.
+     */
+    const incomingMessage = new IncomingMessage(message);
+    // Read the docID string, but don't use it
+    incomingMessage.readVarString();
+
+    // Read the hocuspocusMessageType
+    const hocuspocusMessageType = incomingMessage.readVarUint();
+    // If the hocuspocusMessageType is not Sync, we don't handle the message, since it is not an update
+    if (hocuspocusMessageType !== MessageType.Sync) {
```

Comment:
do we also need `SyncReply`?  https://github.com/ueberdosis/hocuspocus/blob/6fe902a7902e98b6fba3981179ae56b64ad6d4ac/packages/server/src/MessageReceiver.ts#L38

Reply:
Maybe? I'm not familiar with what this bit is doing totally

Reply:
Looks specific to the Redis integration, but doesn't hurt to add: https://github.com/ueberdosis/hocuspocus/blob/05f7980ef8840c20be9145805c466f14b4dd3939/packages/server/src/types.ts#L16